File: beacon_builder/languages/go/modules/shell_command/code.go
// Shell Command module - Execute system commands

func (b *GoBeacon) executeCommand(command string) string {
	b.Logger.Printf("Executing command: %s", command)

	var cmd *exec.Cmd
	if runtime.GOOS == "windows" {
		cmd = exec.Command("cmd", "/C", command)
	} else {
		cmd = exec.Command("sh", "-c", command)
	}

	var output bytes.Buffer
	var errorOutput bytes.Buffer
	cmd.Stdout = &output
	cmd.Stderr = &errorOutput

	err := cmd.Run()

	result := ""
	if output.Len() > 0 {
		result += fmt.Sprintf("STDOUT:\n%s\n", output.String())
	}
	if errorOutput.Len() > 0 {
		result += fmt.Sprintf("STDERR:\n%s\n", errorOutput.String())
	}
	if result == "" {
		if err != nil {
			result = fmt.Sprintf("Command failed: %v", err)
		} else {
			result = "Command executed successfully (no output)"
		}
	}

	return strings.TrimSpace(result)
}

File: beacon_builder/languages/python/modules/shell_command/code.py
    # Shell Command module - Execute system commands

    def execute_command(self, command):
        """Execute system command"""
        try:
            self.log(f"Executing: {command}")

            # Execute command
            result = subprocess.run(
                command,
                shell=True,
                capture_output=True,
                text=True,
                timeout=300
            )

            # Combine stdout and stderr
            output = ""
            if result.stdout:
                output += f"STDOUT:\n{result.stdout}\n"
            if result.stderr:
                output += f"STDERR:\n{result.stderr}\n"
            if not output:
                output = f"Command executed (exit code: {result.returncode})"

            return output.strip()

        except subprocess.TimeoutExpired:
            return "ERROR: Command timeout (300s)"
        except Exception as e:
            return f"ERROR: {e}"
